Fix borgbackup-job -h, --envfile re-introduced

merikz opened the pull request:
envfile is (re-)introduced as required parameter to distinguish
from the (optional) paths-to-backup parameter(s), and to abort
with help text if no envfile is given.

Fixes #9

merikz committed:
Fix borgbackup-job -h, --envfile re-introduced

envfile is (re-)introduced as required parameter to distinguish
from the (optional) paths-to-backup parameter(s), and to abort
with help text if no envfile is given.

Fixes #9

ehannes commented on CHANGELOG.md:
```diff
@@ -5,7 +5,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 ## Unreleased
-
+### Changed
+- Changed syntax: `--envfile <filename>` is now a required parameter
```
Generally, I try to skip the first "Changed" here in "Changed syntax" since we are already under the "Changed" headline. Makes it a bit shorter and direct :smiley_cat:

merikz replied:
ok!

ehannes commented on borgbackup-job:
```diff
@@ -101,26 +93,34 @@ init() {
         die 0
         ;;
-      '--') # Separator between optional and positional parameters
+      --) # Separator between optional and positional parameters
```
This is no longer true. These arguments are no longer optional. We mention this on one other place in this file as well, so we should change both these since we now have required named parameters.

merikz replied:
These '--' is the output by getopt to signal end of getopt parsing. What follows is parameters given by the user that 
getopt does not know about so we 'break' the loop.
Changed the comment.

merikz replied:
not sure which other place you mean

ehannes replied:
Yes, I know what the double dashes are for. What I think we should change is that we call these "optional" parameters. They are no longer optional. This should be changed as well:
```
# Sets double dashes, which marks end of optional parameters
eval set -- "$options"
unset options
```

I can fix it and push...

ehannes reviewed (changes requested):
Nice catch! Much better to use set `--envfile` as required to give better failure messages :+1:

merikz committed:
Changelog updated

merikz committed:
Fix config filenames

Naming convention of generated config files by setup.sh is now:
<reponame>_<repolocation>

Fixes #8

merikz committed:
Silence the linter

merikz committed:
Fix --dry-run

Refactor hooks and --dry-run so hooks are not executed by
--dry-run and the borg command(s) with actual parameter is reported.

Fixes #6

merikz committed:
cleanup: declare local variables as such

variables local to a function should be declared with
'local varname'

ehannes committed:
Rewords "optional paramertes"
